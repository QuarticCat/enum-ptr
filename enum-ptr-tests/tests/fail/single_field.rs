#![allow(dead_code)]

use enum_ptr::EnumPtr;

#[derive(EnumPtr)]
#[repr(C, usize)]
enum Foo<'a, 'b> {
    A(&'a i32),
    B(&'b i32, *const i32),
}

#[derive(EnumPtr)]
#[repr(C, usize)]
enum Bar<'a> {
    A(&'a i32),
    B,
}

fn main() {}
