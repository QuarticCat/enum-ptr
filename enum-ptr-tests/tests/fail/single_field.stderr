error: EnumPtr only supports single field
 --> $DIR/single_field.rs:9:5
  |
9 |     B(&'b i32, *const i32),
  |     ^

error: EnumPtr only supports single field
  --> $DIR/single_field.rs:16:5
   |
16 |     B,
   |     ^
